lib/improver/cli/sleet_probability.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# (C) British Crown Copyright 2017-2019 Met Office.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Script to calculate sleet probability."""
+
+import warnings
+
+from improver import cli
+
+
+@cli.clizefy
+@cli.with_output
+def process(snow: cli.inputcube, rain: cli.inputcube):
+    """Calculate sleet probability.
+
+    Calculates the sleet probability using the
+    calculate_sleet_probability plugin.
+
+    Args:
+        snow (iris.cube.Cube):
+            An iris Cube of the probability of falling snow level at
+            or below surface > threshold (currently 0.8)
+        rain (iris.cube.Cube):
+            An iris Cube of the probability of falling rain level at
+            or above surface > threshold (currently 0.8)
```

Comment:
This might be a question for @MoseleyS, but what does such a cube look like? Currently the CLI test data that has been created has no threshold coordinate; this matches the output from the snow falling level chain

For snow falling level the chain looks like this: 'snow_falling_level', 'perc_nbhood', 'percentiles_to_probabilities', ...

We get altitudes out of the first step. Then we create percentiles from a neighbourbood in the next step. Then we convert these to a probability of snow falling at the surface (our threshold) in the final step. This is a field of probabilities. If Stephen is describing sleet regions as those in which the probability of rain at the surface is < 0.8 and the probability of snow at the surface < 0.8, we need to find these regions, which requires the plugin to do a bit more work.

Reply:
I assumed (from the chain diagrams) that the input going into the sleet cli would have been thresholded already? However perhaps I have got this wrong @MoseleyS ? We don't currently have both prob(snow_falling_level) > 0.8 and prob(rain_falling_level) > 0.8 being produced so I just tried to set up something that will work once we have both of those?

Reply:
I think Katie is right. There will be an instance of the threshold plugin between the percentiles-to-probabilities CLI and this CLI to make the values categorical. The input cubes should have names like "probability_of_altitude_of_snow_falling_level_below_surface" and "probability_of_altitude_of_rain_falling_level_above_surface". Both cubes will have a scalar coord with var_name="threshold".
The more I think about this, the more I think there might be a potential meta-data minefield. I will try to write down what I think the cubes from each step will look like.

Reply:
Yes @MoseleyS  I was thinking about the sleet cube and what it should be called too. I have just called it "probability_of_sleet" at the moment, but potentially it would want to be called "probability_of_altitude_of_sleet_falling_level_below_surface" or something else once we know what the metadata of the input cubes should look like. At the moment the plugin just uses the attributes, etc of the input snow cube and changes the name.

Reply:
Okay, I had missed the extra threshold step in the chain, which means we should have a 2D field of probabilities for the rain and snow as required.

Reply:
I have emailed @MoseleyS  and @fionaRust and we have discussed the "threshold" step before this plugin/cli. The existing threshold step doesn't do exactly what we want and so there will need to be another ticket to do this. However with changes to pytest, etc we have decided it should be a separate piece of work rather than done within this ticket.